modules/javafx.graphics/src/main/java/javafx/scene/Node.java

```diff
@@ -1397,6 +1398,87 @@ public String getName() {
         return visible;
     }
 
+    /**
+     * Whether or not this {@code Node} is showing (that is, it is part of an
+     * open Window on the user's system). The Window might be "showing", yet the user might not
+     * be able to see it due to the Window being rendered behind another Window
+     * or due to the Window being positioned off the monitor.
+     *
+     * <p>Note that the {@code Node} does not need to be visible for this property
+     * to be {@code true}.
+     *
+     * @defaultValue false
+     * @since 20
+     */
+    private ReadOnlyBooleanProperty shown;
+
+    public final boolean isShown() {
+        Scene s = getScene();
+        if (s == null) return false;
+        Window w = s.getWindow();
+        return w != null && w.isShowing();
+    }
+
+    public final ReadOnlyBooleanProperty shownProperty() {
```

Comment:
Discussion: Note I had to wrap the ObservableValue<Boolean> to become a ReadOnlyBooleanProperty, even though ObservableValue is already read-only and quite usable without the wrapper. I would have preferred to just have the shown property signature be:

    ObservableValue<Boolean> shownProperty();

Or even better:

    ObservableValue<Boolean> shown();

However, I think it would not be detected as a JavaFX property then, and it would also break with the tradition that properties are, well, properties. Thoughts?